SWLOR.Game.Server/Native/GetDamageRoll.cs

```diff
@@ -188,6 +195,24 @@ private static int OnGetDamageRoll(void* thisPtr, void* pTarget, int bOffHand, i
                 dmgValues[CombatDamageType.Physical] += 6;
             }
 
+            // MGT mod bonus to Crushing Staves + Strong Style
+            var mightMod = attacker.m_pStats.m_nStrengthModifier;
+            var playerId = attacker.m_pUUID.GetOrAssignRandom().ToString();
+
+            if (Item.StaffBaseItemTypes.Contains((BaseItem)weapon.m_nBaseItem))
+            {
+                if (attacker.m_pStats.HasFeat((ushort)FeatType.CrushingMastery) == 1)
+                    dmgValues[CombatDamageType.Physical] += mightMod * 2; // Mastery gives 2x MGT
+                else if (attacker.m_pStats.HasFeat((ushort)FeatType.CrushingStyle) == 1)
+                    dmgValues[CombatDamageType.Physical] += mightMod; // Crushing Staves 1x MGT
+            }
+            else if (Item.SaberstaffBaseItemTypes.Contains((BaseItem)weapon.m_nBaseItem) && 
+                Ability.IsAbilityToggled(playerId, AbilityToggleType.StrongStyleSaberstaff))
+                dmgValues[CombatDamageType.Physical] += (int)Math.Ceiling(mightMod / 2.0f);
+            else if (Item.LightsaberBaseItemTypes.Contains((BaseItem)weapon.m_nBaseItem) &&
+                Ability.IsAbilityToggled(playerId, AbilityToggleType.StrongStyleLightsaber))
+                dmgValues[CombatDamageType.Physical] += (int)Math.Ceiling(mightMod / 2.0f);
+
             // Doublehand perk
             if (weapon != null)
```

Comment:
After you fix the null check above, you can probably remove the null check here.